peer/peer.go

```diff
@@ -2215,7 +2270,39 @@ func (p *Peer) waitToFinishNegotiation(pver uint32) error {
 //     that btcd does not implement but bitcoind does.
 //  6. If remote peer sent sendaddrv2 above, wait until receipt of verack.
 func (p *Peer) negotiateInboundProtocol() error {
-	if err := p.readRemoteVersionMsg(); err != nil {
+	// We may be anticipating a v2 connection, but if the initiating peer sends
+	// us a v1 version message, we need to note down that we've downgraded the
+	// connection internally.
+	downgradedConn := false
+
+	if p.cfg.UsingV2Conn {
+		// TODO: Change garbageLen to be random.
```

Comment:
So just a random 12 byte int?

Reply:
I'm not sure why I chose it to be `1` here, it can instead be `0` I think. The garbage here is limited to `4095` bytes.